lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -50,399 +37,218 @@
-        """Returns true if plugin is installed."""
-        return plugin.name in self._installed_plugins
-
-    def _user_requested_to_enable(self, plugin: OpenSearchPlugin) -> bool:
-        """Returns True if user requested plugin to be enabled."""
-        return plugin.requested_to_enable()
-
-    def _is_enabled(self, plugin: OpenSearchPlugin) -> bool:
-        """Returns true if plugin is enabled.
-
-        The main question to answer is if we would have it in the configuration
-        from cluster settings. If yes, then we know that the service is enabled.
-
-        Check if the configuration from enable() is present or not.
-
-        Raise:
-            OpenSearchKeystoreNotReadyError: If the keystore is not yet ready
-        """
-        # Avoid the keystore check as we may just be writing configuration in the files
-        # while the cluster is not up and running yet.
-        if plugin.config().secret_entries:
-            # Need to check keystore
-            # If the keystore is not yet set, then an exception will be raised here
-            keys_available = self._keystore.list
-            keys_to_add = [
-                k
-                for k in plugin.config().secret_entries.keys()
-                if plugin.config().secret_entries.get(k)
-            ]
-            if any(k not in keys_available for k in keys_to_add):
-                return False
-            keys_to_del = [
-                k
-                for k in plugin.config().secret_entries.keys()
-                if not plugin.config().secret_entries.get(k)
-            ]
-            if any(k in keys_available for k in keys_to_del):
-                return False
-
-        # We always check the configuration files, as we always persist data there
-        existing_setup = self._opensearch_config.get_plugin(plugin.config().config_entries)
-
-        if any([k not in existing_setup.keys() for k in plugin.config().config_entries.keys()]):
-            return False
-
-        # Now, we know the keys are there, we must check their values
-        return all(
-            [
-                plugin.config().config_entries[k] == existing_setup[k]
-                for k in plugin.config().config_entries.keys()
-            ]
+            # if unit is main orchestrator leader, remove secrets transferred to subclusters
+            if self.charm.opensearch_peer_cm.is_provider(typ="main"):
+                self.charm.peer_cluster_provider.refresh_relation_data(event)
+        except SecretNotFoundError:
+            logger.debug("Can't find secret %s", self.secret_label)
+
+    def _on_secret_changed(self, event) -> None:
+        """Handles secret changes"""
+        label = self.charm.secrets.label(Scope.APP, self.secret_label)
+        if label != event.secret.label:
+            return
+
+        secret = event.secret.get_content(refresh=True)
+        raw = secret.get(self.secret_label)
+        keys = json.loads(raw)
```

Comment:
We can do it  more gracefully to prevent unexpected errors.
```suggestion
        content = event.secret.get_content(refresh=True)
        raw = content.get(self.secret_label)
        if not raw:
            logger.warning("Secret %s has no %s payload", event.secret.label, self.secret_label)
            return

        try:
            keys = json.loads(raw)
        except json.JSONDecodeError:
            logger.error("Malformed JSON in secret %s", event.secret.label)
            return
```